boot-speed/clouds/measure-cloud.py

```diff
@@ -15,13 +15,66 @@
 import os
 import paramiko
 import pycloudlib
+import shutil
 import sys
 
 from pathlib import Path
 
 
 known_clouds = ['ec2', 'gce']
 
+class EC2Instspec():
+    def __init__(self, release):
+        self.release = release
+
+    # Defaults
+    region = "us-east-1"
+    inst_type = "t2.micro"
+    subnetid = ""
+    sgid = []
+
+    def measure(self, datadir, instances=1, reboots=1):
+        """
+        Measure Amazon AWS EC2.
+        Returns the measurement metadata as a dictionary
+        """
+        print('Perforforming measurement on Amazon EC2')
+
+        logging.basicConfig(level=logging.DEBUG)
+        tag = "bootspeed-" + str(int(dt.datetime.utcnow().timestamp()))
+        ec2 = pycloudlib.EC2(tag=tag, region=self.region)
+
+        if self.inst_type.split('.')[0] == 'a1':
+            daily = ec2.daily_image(release=self.release, arch='arm64')
+        else:
+            daily = ec2.daily_image(release=self.release)
+
+        print("Daily image for", self.release, "is", daily)
+
+        metadata = gen_metadata("ec2", self.region, self.inst_type, self.release, daily)
+
+        for ninstance in range(instances):
+            instance_data = Path(datadir, "instance_" + str(ninstance))
+            instance_data.mkdir()
+
+            print("Launching instance", ninstance+1, "of", instances)
+            instance = ec2.launch(
+                daily,
+                instance_type=self.inst_type,
+                SubnetId=self.subnetid,
+                SecurityGroupIds=self.sgid
+            )
+            print("Instance launched.")
+
+            try:
+                measure_instance(instance, instance_data, reboots)
+            finally:
+                print("Deleting the instance.")
+                instance.delete(wait=False)
```

Comment:
As we don't wait for `delete`, do we run in to any problems with reusing the same name/tag when running more than a single instance?

Reply:
I now make sure that every instance has a different tag. I think we are going to run only one instance per job, but you are right: if running more than one this could have caused some headaches. Thanks!